src/parser/shared/normalizers/Channeling.ts

```diff
@@ -0,0 +1,344 @@
+import SPELLS from 'common/SPELLS';
+import CASTS_THAT_ARENT_CASTS from 'parser/core/CASTS_THAT_ARENT_CASTS';
+import {
+  AnyEvent,
+  BeginCastEvent,
+  BeginChannelEvent,
+  CastEvent,
+  EndChannelEvent,
+  EventType,
+  HasAbility,
+  HasSource,
+} from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import InsertableEventsWrapper from 'parser/core/InsertableEventsWrapper';
+import { Options } from 'parser/core/Module';
+
+/**
+ * Channels and casts are handled differently in events, and some information is also missing and must be inferred.
+ * This normalizer fabricates new BeginChannel and EndChannel events that delineate the cast times
+ * for both casts and channels, and additionally includes inferred information.
+ *
+ * Instant spells have only a Cast event - they are ignored by this normalizer, which
+ * is only concerned with cast time abilities.
+ *
+ * Spells with a cast time have a BeginCast event when they are started and and a Cast event when
+ * they finish. If they are cancelled before finishing, there will be a BeginCast without the linked
+ * Cast. For completed casts, this normalizer fabricates a BeginChannel for the start of the spell
+ * and an EndChannel for the finish. For cancelled casts, there will only be a BeginChannel event
+ * with the 'isCancelled' flag set.
+ *
+ * Channelled spells are handled inconsistently in the events, but this normalizer aims to uniformly
+ * represent them with a BeginChannel when started and and EndChannel when stopped (regardless of if
+ * the spell finished or it was stopped early). There are a variety of ways channeled spells show
+ * in events, and this normalizer allows special case handling to be registered for each.
+ */
+class Channeling extends EventsNormalizer {
+  /**
+   * Listing of all special case handlers for channels
+   */
+  static CHANNEL_SPECS: ChannelSpec[] = [
```

Comment:
I wonder if we could have this defined in the spec's analysis area ie (analysis\mnokmistweaver\normalizers\MWChanneling.ts)

I just have never been a fan of big master lists so its w/e either way

Reply:
I definitely was considering it, and could change it to be that way. I mostly have it in master list form because it was easier to mock up that way. I plan to at least make `CHANNEL_SPECS` passed in instead of static so it's extensible.